org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java

```diff
@@ -0,0 +1,201 @@
+		return c;
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link ServiceReference}.
+	 * Checking if type of {@link ServiceReference} <b>matches</b> the given
+	 * Class<?> objectClass matches a objectClass of the ServiceReference.
+	 *
+	 * <pre>
+	 * List<ServiceReference<?>> serviceEvents = null;
+	 *
+	 * static void example_objectClass(Class<?> objectClass) {
+	 *
+	 * 	assertThat(serviceEvents)// created an {@link ListAssert}
+	 * 		.have(objectClass(objectClass))
+	 * 		.filteredOn(objectClass(objectClass))
+	 * 		.first()// map to {@link // ObjectAssert}
+	 * 		.has(objectClass(objectClass));// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param objectClass - the objectClass that would be tested against the
+	 *            ServiceReference
+	 * @return the Condition
+	 */
+
+	static Condition<ServiceReference<?>> objectClass(final Class<?> objectClass) {
+		return new Condition<ServiceReference<?>>(sr -> {
+			Object classes = sr.getProperty(Constants.OBJECTCLASS);
+			if (classes != null && classes instanceof String[]) {
+				return Stream.of((String[]) classes)
+					.filter(Objects::nonNull)
+					.anyMatch(objectClass.getName()::equals);
+			}
+			return false;
+		}, "has Objectclass %s", objectClass.getName());
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link ServiceReference}.
+	 * Checking if the serviceProperties of the {@link ServiceReference}
+	 * <b>matches</b> the given Condition<Dictionary<String, Object>>.
+	 *
+	 * <pre>
+	 * List<ServiceReference> serviceReference = null;
+	 *
+	 * static void servicePropertiesHas(Condition<Dictionary<String,Object>> condition)) {
+	 *
+	 * assertThat(serviceReference)// created an {@link ListAssert}
+	 *   .have(servicePropertiesHas(condition))
+	 *   .filteredOn(servicePropertiesHas(condition))
+	 *   .first()// map to {@link ObjectAssert}
+	 *   .has(servicePropertiesHas(condition));// used on {@link
+	 ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<ServiceReference<?>> servicePropertiesHas(Condition<Dictionary<String, Object>> condition) {
+		return MappedCondition.mappedCondition(sr -> Dictionaries.asDictionary(sr), condition,
+			"ServiceReference to Dictionary");
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link ServiceReference}.
+	 * Checking if the serviceProperties of the {@link ServiceReference} <b>is
+	 * not null</b>.
+	 *
+	 * <pre>
+	 * List<ServiceReference> serviceReference = null;
+	 *
+	 * static void example_servicePropertiesIsNotNull() {
+	 *
+	 * 	assertThat(serviceReference)// created an {@link ListAssert}
+	 * 		.have(servicePropertiesIsNotNull())
+	 * 		.filteredOn(servicePropertiesIsNotNull())
+	 * 		.first()// map to {@link
+	 * 				// ObjectAssert}
+	 * 		.has(servicePropertiesIsNotNull());// used on {@link
+	 * 											// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+
+	/**
```

Comment:
This appears unfinished. TODOs

Reply:
this is prepared for the moment when we rely on R7. The methods does not exist in R6

Reply:
OK, but then the javadoc comment above is not commenting on an extant member.